pkg/registry/daemonset/strategy.go

```diff
@@ -99,6 +99,14 @@ func (daemonSetStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object
 	return append(validationErrorList, updateErrorList...)
 }
 
+func (daemonSetStrategy) SupportGraceful() bool {
```

Comment:
I'm saying define a `type GracefulDeleter interface {
GracefulDeletion(blah) }` and check to see if the delete strategy
implements that instead of adding the SupportGraceful() function.

On Tue, Mar 15, 2016 at 4:15 PM, Chao Xu <notifications@github.com> wrote:

> In pkg/registry/daemonset/strategy.go
> <https://github.com/kubernetes/kubernetes/pull/22965#discussion_r56257559>
> :
>
> > @@ -99,6 +99,14 @@ func (daemonSetStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object
> >  	return append(validationErrorList, updateErrorList...)
> >  }
> >
> > +func (daemonSetStrategy) SupportGraceful() bool {
>
> We need a runtime.Typer to get the ObjectMeta of a runtime.Object. I'm
> plumbing it throuhg DeleteStrategy.
>
> Regarding why I added SupportGraceful(), it's because previously the
> BeforeDelete() quickly returns if the DeleteStrategy is nil, but now I
> force DeleteStrategy to be non-nil, so we need to add SupportGraceful() to
> return quickly.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/22965/files/754caf176c23fe105cf47de2eb51aa89753e325a#r56257559>
>